### PR TITLE
almost: complete hackerblocks intergration

### PR DESCRIPTION
**Pull request by abhishek97**

### CODE REVIEW

File: app/pods/components/player-code-challenge/component.js [hunk 1]
```diff
@@ -1,6 +1,162 @@
...
+            console.log(result)
+            this.set('problemJsonApiPayload', result)
+        }).catch(err => {
+            console.log(err)
+        })
+    }
+
+    @action
+    runCode () {
+        this.get('ajax').request('https://judge.cb.lk/api/submission', {
```
Comment (witty123) [inline app/pods/components/player-code-challenge/component.js]: remove this integration please. This is a synchronous call which will be deprecated soon. We don't want to block our judge for a request from amoeba particularly when everything else is working asynchronously.

File: app/pods/components/player-code-challenge/component.js [hunk 2]
```diff
@@ -1,6 +1,162 @@
...
+    didReceiveAttrs () {
+        this._super(...arguments)
+        const code = this.get('code')
+        this.get('hbApi').request('problems', {
+            data: {
+                contest_id: code.get('hbContestId'),
+                problem_id: code.get('hbProblemId')
+            }
+        }).then(result => {
+            console.log(result)
```
Comment (witty123) [inline app/pods/components/player-code-challenge/component.js]: remove console logs

File: app/pods/components/player-code-challenge/component.js [hunk 3]
```diff
@@ -1,6 +1,162 @@
...
+import { service } from 'ember-decorators/service'
+import { action } from 'ember-decorators/object'
+import { computed } from 'ember-decorators/object'
+import config from 'codingblocks-online/config/environment'
 
-export default Component.extend({
-    code: computed.alias('payload')
-});
+
+export default class CodeChallengeComponent extends Component {
```
Comment (witty123) [inline app/pods/components/player-code-challenge/component.js]: write logic for getting existing submissions along with leaderboard.

**Comment by witty123** [comment]
@abhishek97 too many scss changes here, would be removed once you take the latest pull of motley, approve these changes in motley first and then use those changes here.

**Comment by abhishek97** [comment]
Those scss changes are result of a pull from motley. Once master has updated motley, those will not show here. I don't change any case here.

**Pull request merged**